Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -325,54 +336,72 @@ namespace AzToolsFramework
-    //! otherwise create a new pair of the first widget in the shared column, with size of 2 elements.
-    void DPELayout::SharePriorColumn(QWidget* headWidget)
+    void DPELayout::AddSharePriorColumn(size_t previousIndex, size_t widgetIndex)
     {
-        if (ShouldSharePrior())
+        // Add to an existing sharePrior group if the previous widget's index is already there, otherwise create a new group
+        if (!m_sharePriorColumn.empty() && m_sharePriorColumn.back().back() == previousIndex)
         {
-            int newWidgetCount = m_sharePriorColumn.back().second + 1;
-            m_sharePriorColumn[m_sharePriorColumn.size() - 1].second = newWidgetCount;
+            m_sharePriorColumn.back().push_back(widgetIndex);
         }
         else
         {
-            auto widgetColumnPair = AZStd::pair<QWidget*, int>(headWidget, 2);
-            m_sharePriorColumn.push_back(widgetColumnPair);
+            AZStd::vector<size_t> newEntry;
+            newEntry.push_back(previousIndex);
+            newEntry.push_back(widgetIndex);
+            m_sharePriorColumn.push_back(newEntry);
         }
     }
 
-    void DPELayout::SetSharePrior(bool sharePrior)
-    {
-        m_shouldSharePrior = sharePrior;
-    }
-
-    bool DPELayout::ShouldSharePrior()
+    void DPELayout::RemoveSharePriorColumn(size_t widgetIndex)
     {
-        return m_shouldSharePrior;
+        for (auto groupIt = m_sharePriorColumn.begin(); groupIt != m_sharePriorColumn.end(); ++groupIt)
+        {
+            AZStd::vector<size_t> currentGroup = *groupIt;
+            for (int currentGroupIndex = 0; currentGroupIndex < currentGroup.size(); currentGroupIndex++)
+            {
+                if (widgetIndex == currentGroup[currentGroupIndex])
+                {
+                    if (currentGroupIndex == currentGroup.size() - 1)
+                    {
+                        // If we are removing from shared group of 2 or less, erase the group
+                        if (currentGroup.size() <= 2)
+                        {
+                            m_sharePriorColumn.erase(groupIt);
+                            return;
+                        }
+                        // Group size is bigger than 2 and widget is at the end of the group, so just remove it
+                        else
+                        {
+                            currentGroup.erase(currentGroup.begin() + currentGroupIndex);
+                            return;
+                        }
+                    }
+                    // If the widget is the second member of the group, remove the first widget in the group
+                    else if (currentGroupIndex == 1)
+                    {
+                        currentGroup.erase(currentGroup.begin());
+                        return;
+                    }
+                    else
+                    {
+                        //! to-do: handle case where we remove widget from the middle of a share prior group
```

Comment:
This is the case that needs to be handled for shared groups larger than 4 elements, because if the second element or third element is removed from this group, then we need to update behavior according to what the user would expect to happen.

Reply:
I'd like to discuss it, but I have an idea in theory of what we could do here
This is really important to do correctly for shared groups to scale, but for now I left it since I haven't seen anything in the inspector/asset browser have more than 3 elements in a shared group